pkg/sql/telemetry_logging_test.go

```diff
@@ -976,6 +976,115 @@ func TestTelemetryLoggingPerFingerprintSampling(t *testing.T) {
 	}
 }
 
+// TestTelemetryLoggingInternalEnabled verifies that setting the cluster setting to send
+// internal queries to telemetry works as intended.
+func TestTelemetryLoggingInternalEnabled(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	sc := log.ScopeWithoutShowLogs(t)
+	defer sc.Close(t)
+
+	cleanup := logtestutils.InstallLogFileSink(sc, t, logpb.Channel_TELEMETRY)
+	defer cleanup()
+
+	st := logtestutils.StubTime{}
+	sts := logtestutils.StubTracingStatus{}
+
+	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			EventLog: &EventLogTestingKnobs{
+				// The sampling checks below need to have a deterministic
+				// number of statements run by internal executor.
+				SyncWrites: true,
+			},
+			TelemetryLoggingKnobs: &TelemetryLoggingTestingKnobs{
+				getTimeNow:       st.TimeNow,
+				getTracingStatus: sts.TracingStatus,
+			},
+		},
+	})
+
+	defer s.Stopper().Stop(context.Background())
+
+	db := sqlutils.MakeSQLRunner(sqlDB)
+	db.Exec(t, `SET application_name = 'telemetry-internal-logging-test'`)
+
+	db.Exec(t, `SET CLUSTER SETTING sql.telemetry.query_sampling.max_event_frequency = 10;`)
+	db.Exec(t, `SET CLUSTER SETTING sql.telemetry.query_sampling.enabled = true;`)
+	db.Exec(t, `SET CLUSTER SETTING sql.telemetry.query_sampling.internal.enabled = true;`)
+
+	// This query should trigger 2 internal TRUNCATE queries.
+	// Since TRUNCATE is not a DML stmt, both should certainly get logged.
+	// Meanwhile, the SELECT query itself should not be logged due to the above
+	// query being logged first.
+	stubTime := timeutil.FromUnixMicros(int64(1e6))
+	st.SetTime(stubTime)
+	_, err := db.DB.ExecContext(context.Background(), `SELECT crdb_internal.reset_sql_stats();`)
+	if err != nil {
+		t.Errorf("unexpected error executing query: %v", err)
+	}
+
+	expectedQueries := []struct {
+		logMsg     string
+		isInternal bool
+	}{
+		{
+			logMsg:     `TRUNCATE TABLE system.public.statement_statistics`,
+			isInternal: true,
+		},
+		{
+			logMsg:     `TRUNCATE TABLE system.public.transaction_statistics`,
+			isInternal: true,
+		},
+	}
+
+	log.Flush()
+
+	entries, err := log.FetchEntriesFromFiles(
+		0,
+		math.MaxInt64,
+		10000,
+		regexp.MustCompile(`"EventType":"sampled_query"`),
+		log.WithMarkedSensitiveData,
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(entries) == 0 {
+		t.Fatal(errors.Newf("no entries found"))
+	}
+
+	for _, e := range entries {
+		if strings.Contains(e.Message, "SELECT crdb_internal.reset_sql_stats") {
+			t.Errorf("unexpected query log: %s", e.Message)
+		}
+	}
+
+	// Attempt to find all expected logs.
+	for _, expected := range expectedQueries {
+		found := false
+		execMode := executorTypeExec
+		if expected.isInternal {
```

Comment:
A little confused as to why this is added here since all the `expectedQueries` in this test are internal. Are you planning on adding more queries with `isInternal: false`?

Reply:
Removed. Yeah originally I was going to include non-internal queries but with allowing internal queries I couldn't be sure there wasn't an internal query running right before executing the user query.